leases/schema/mutations.py

```diff
@@ -419,42 +419,21 @@ class Input(SendExistingInvoicesInput):
             required=True, description="API token for Helsinki profile GraphQL API",
         )
 
-    result = graphene.Field(SendExistingInvoicesType)
+    ok = graphene.Boolean(required=True)
 
     @classmethod
     @view_permission_required(CustomerProfile)
     @change_permission_required(BerthLease, WinterStorageLease, Order)
     def mutate_and_get_payload(cls, root, info, profile_token, **input):
-        from payments.schema import OrderNode
-
-        try:
-            result = BerthInvoicingService(
-                request=info.context,
-                profile_token=profile_token,
-                due_date=input.get("due_date"),
-            ).send_invoices()
-        except AutomaticInvoicingError as e:
-            raise VenepaikkaGraphQLError(e)
-
-        # Only need to parse the list of ids
-        successful_orders = [
-            to_global_id(OrderNode, id) for id in result.get("successful_orders")
-        ]
-        # Parse the failed object types
-        failed_leases = list(
-            map(parse_invoicing_result(BerthLeaseNode), result.get("failed_leases"))
-        )
-        failed_orders = list(
-            map(parse_invoicing_result(OrderNode), result.get("failed_orders"))
-        )
-
-        result_response = SendExistingInvoicesType(
-            successful_orders=successful_orders,
-            failed_orders=failed_orders,
-            failed_leases=failed_leases,
+        service = BerthInvoicingService(
+            request=info.context,
+            profile_token=profile_token,
+            due_date=input.get("due_date"),
         )
+        t1 = threading.Thread(target=service.send_invoices, args=[])
```

Comment:
If I remember correctly, the currently active language in Django is thread-specific, so if translations are needed, a translation.activate() call is needed in send_invoices. How is the language for "email to admins" determined? Does it use currently active translation or does it get it from elsewhere?

Reply:
It takes the default language from the settings, which is Finnish. Since the admins are only Finnish-speaking, that's not a concern. 👍